libdrgn/program.c

```diff
@@ -655,9 +672,49 @@ static struct drgn_error *get_prstatus_pid(struct drgn_program *prog, const char
 	return NULL;
 }
 
-struct drgn_error *drgn_program_cache_prstatus_entry(struct drgn_program *prog,
-						     const char *data,
-						     size_t size)
+struct drgn_error *drgn_thread_dup_internal(const struct drgn_thread *thread,
+					    struct drgn_thread *ret)
+{
+	struct drgn_error *err = NULL;
+	ret->tid = thread->tid;
+	ret->prog = thread->prog;
+	if (thread->prog->flags &
+	    (DRGN_PROGRAM_IS_LINUX_KERNEL | DRGN_PROGRAM_IS_LIVE)) {
```

Comment:
The idea was that if we eventually add support for live userspace processes, this wouldn't need to be changed (because we'd also be defining the object there and pointing it at a `pthread` or something).

Reply:
I see. The `pthread` handle would be just as applicable to userspace core dumps, which is why I didn't realize what this was going for.

Reply:
I didn't realize that. I'll change it to just check if it's the kernel for the time being then.